src/__tests__/pages/api/auth/reset-password.test.ts

```diff
@@ -0,0 +1,299 @@
+import dotenv from 'dotenv';
+dotenv.config();
+
+import {
+  mockUserData,
+  mockPassword,
+  mockEmail
+} from '../../../../__mocks__/pages/api/auth';
+import signUpHandler from '../../../../pages/api/auth/signup';
+import resetPasswordHandler from '../../../../pages/api/auth/reset-password';
+import { PrismaClient } from '@prisma/client';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { hashFunction } from '../../../../utils/password';
+import { resetPasswordProps } from '../../../../types/api/auth/reset-password';
+
+const prisma = new PrismaClient();
+
+let email: string, password: string;
+
+beforeEach(async () => {
+  const userData = mockUserData();
+  email = userData.email;
+  password = userData.password;
+
+  const req = {
+    body: {
+      email,
+      password
+    }
+  } as NextApiRequest;
+
+  const json = jest.fn();
+
+  const status = jest.fn(() => {
+    return { json };
+  });
+
+  const res = {
+    status
+  } as unknown as NextApiResponse;
+
+  await signUpHandler(req, res);
+});
+
+describe('Successfully change password', () => {
+  it('Change password successs', async () => {
+    const hashPassword = (
+      await prisma.user.findUnique({
```

Comment:
Logically this breaks if sign-up breaks though, right? When we sign up the mock user in the `beforeEach`, we should do it manually without the API endpoint that does it